### ISSUE TITLE
Make the API layer more conventional

### ISSUE THREAD

**Issue by hrshdhgd** [mapping-commons/sssom-py#376]
For e.g.:
`parse_file` conventionally takes as input and returns object(s) but in our case, it takes a path, converts the data into SSSOM format and exports it (stdout or output file if path provided.).

**Comment by matentzn** [comment]
I agree, we should align the design here a bit with pandas.read_csv() etc
